skrub/_drop_null.py

```diff
@@ -0,0 +1,54 @@
+# drop columns that contain all null values
+from sklearn.utils.validation import check_is_fitted
+
+from . import _dataframe as sbd
+from ._on_each_column import SingleColumnTransformer
+
+__all__ = ["DropNullColumn"]
+
+
+class DropNullColumn(SingleColumnTransformer):
+    """Drop a single column if it contains only null values."""
+
+    def __init__(self):
+        super().__init__()
+        self._is_fitted = False
+
+    def fit_transform(self, column, y=None):
+        """Fit the encoder and transform a column.
+
+        Args:
+            column : Pandas or Polars series. The input column to check.
+            y : None. Ignored.
+
+        Returns:
+            The input column, or an empty list if the column contains only null values.
+        """
+        del y
+
+        self._is_fitted = True
+        return self.transform(column)
```

Comment:
we want the schema of the output to be the same after every call to `transform`. at the moment, the check is done during transform so we decide each time transform is called, depending on the input, if we want to drop it or not. instead, we want to check if the column passed to `fit` is all null, if yes, we always drop during transform (even if we later get a column with non-null values) if no, we never drop (even if we later get a column full of nulls).

so it would look like:

```python
def fit_transform(self, column, y=None):
    self.drop_ = sbd.is_all_null(column)
    return self.transform(column)

def transform(self, column):
    check_is_fitted(self, "drop_")
    return [] if self.drop_ else column

# __sklearn_is_fitted__ not needed anymore
```